macros/code-hasher/src/lib.rs

```diff
@@ -0,0 +1,58 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+//
+// Copyright (c) DUSK NETWORK. All rights reserved.
+
+//! Tiny proc macro library designed to hash a code block generating a unique
+//! identifier for it which will get written into a `const` inside of the code
+//! block.
+//!
+//! # Example
+//! extern crate code_hahser;
+//! ```
+//! #[code_hasher::hash(SOME_CONST_NAME, version = "0.1.0")]
+//! pub mod testing_module {
+//!
+//!     pub fn this_does_something() -> [u8; 32] {
+//!         SOME_CONST_NAME
+//!     }
+//! }
+//! ```
+//! Here, `SOME_CONST_NAME` has assigned as value the resulting hash of:
+//! - The code contained inside `testing_module`.
+//! - The version of the crate (code_hasher) version.
+//! - The version passed by the user (is optional). Not adding it will basically
+//!   not hash this attribute and **WILL NOT** use any default alternatives.
+
+use blake3::Hasher;
+use proc_macro::TokenStream;
+
+#[proc_macro_attribute]
+pub fn hash(_attr: TokenStream, _input: TokenStream) -> TokenStream {
```

Comment:
Nit: You're using both arguments now, so you should remove the underscore as prefix (so it should be `attr` and `input` instead of `_attr` and `_input`.